pagopa/io-app
# chore: [IOBP-319] Add new wallet payment redux and saga

## Short description
This PR introduces the reducer and networking sagas to handle the new wallet payment flow.
The flow covers only the happy path and displays the store content in the screens.
**Note:** the authorization process is not covered by this implementation and will be handled in [IOBP-432](https://pagopa.atlassian.net/browse/IOBP-432)

https://github.com/pagopa/io-app/assets/6160324/201e42e4-13ad-465c-a5a3-8bd517c11cb2


## List of changes proposed in this pull request
- Added Wallet eCommerce API definitons url and script in `package.json`
- Added `DebugPrettyPrint`, an utility component to display the content of any object in a pleasant and readable way.
- Renamed `WALLET_API_BASEURL` and `WALLET_API_UAT_BASEURL` in `.env.local`, `.env.production` and `config.ts` files
- Moved barcode scan files from `ts/features/walletV3/payment` into dedicated subfolder in `ts/features/walletV3/barcode`
- Added client, saga and reducer for Wallet eCommerce
- Added barebone logic into wallet payment flow

## How to test
Go to **Profile > New Wallet Playground > Payment Playground** using the `io-dev-api-server` as the backend. Just make sure to checkout this branch: https://github.com/pagopa/io-dev-api-server/pull/322 and set the following in the config.json

```json
  "wallet": {
    "useLegacyRptIdVerificationSystem": true
  }
```


[IOBP-432]: https://pagopa.atlassian.net/browse/IOBP-432?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
